x-pack/plugins/snapshot_restore/server/types.ts

```diff
@@ -31,4 +31,20 @@ export interface RouteDependencies {
   };
 }
 
+/**
+ * An object representing a resolved index, data stream or alias
+ */
+interface DataObjectFromES {
+  name: string;
+  // per https://github.com/elastic/elasticsearch/pull/57626
+  attributes: Array<'open' | 'closed' | 'hidden' | 'frozen'>;
+  data_stream?: string;
+}
+
+export interface ResolveIndexResponseFromES {
+  indices: DataObjectFromES[];
+  aliases: DataObjectFromES[];
+  data_streams: Array<{ name: string; backing_indices: string[]; timestamp_field: string }>;
```

Comment:
This isn't typed as `DataObjectFromES` but that type's comment mentions data streams (ln 35). Maybe we should remove the mention of data stream from that comment?